ntc_rosetta/parsers/openconfig/junos/__init__.py

```diff
@@ -18,9 +19,15 @@ class JunosParser(parser.RootParser):
 
     class Yangify(parser.ParserData):
         def init(self) -> None:
-            self.root_native["dev_conf"] = etree.fromstring(
-                self.root_native["dev_conf"]
-            )
+            parsed_xml = etree.fromstring(self.root_native["dev_conf"])
+            if parsed_xml.tag != "configuration":
+                parsed_xml = parsed_xml.find('configuration')
+                if parsed_xml is None: 
+                    raise AttributeError("Unable to locate 'configuration' tag in XML blob")
+                # We need to copy the XML element here, otherwise it will retain its parent 
+                # references, meaning we can inadvertantly walk up the tree when we shouldn't.
+                parsed_xml = copy(parsed_xml)
```

Comment:
You may need to do a `copy.deepcopy()` here since `copy.copy()` is only a shallow copy (first level), meaning nested objects/dicts/lists are not duplicated.

Reply:
Updated
I did test both copy and deepcopy, and both appeared to work correctly. At the time I figured we might as well go with the faster copy.
Upon further reflection, you're right, as there may be unexpected ramifications down the road when doing a shallow copy.

Reply:
👍